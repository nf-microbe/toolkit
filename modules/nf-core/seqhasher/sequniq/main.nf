process SEQHASHER_SEQUNIQ {
    tag "${meta.id}"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/gawk:5.3.0':
        'biocontainers/gawk:5.3.0' }"

    input:
    tuple val(meta), path(seq_hasher_tsv)

    output:
    tuple val(meta), path("${prefix}.dups.tsv")     , optional: true    , emit: dups
    tuple val(meta), path("${prefix}.seq-uniq.tsv")                     , emit: unique
    path "versions.yml"                                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    awk 'seen[\$2]++ {
        print \$1 > "${prefix}.dups.tsv"; next
    }
    {
        print \$1 > "${prefix}.seq-uniq.tsv"
    }' ${seq_hasher_tsv}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        awk: \$( awk --version| head -n 1 | sed 's/GNU Awk //; s/, .*//' )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.dups.tsv
    touch ${prefix}.seq-uniq.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        awk: \$( awk --version| head -n 1 | sed 's/GNU Awk //; s/, .*//' )
    END_VERSIONS
    """
}
