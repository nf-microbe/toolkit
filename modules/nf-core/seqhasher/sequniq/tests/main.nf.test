nextflow_process {

    name "Test Process SEQHASHER_SEQUNIQ"
    script "../main.nf"
    process "SEQHASHER_SEQUNIQ"

    tag "modules"
    tag "modules_nfcore"
    tag "cat"
    tag "cat/cat"
    tag "seqhasher"
    tag "seqhasher/seqhasher"
    tag "seqhasher/sequniq"

    setup {
        run("CAT_CAT") {
            script "../../../cat/cat/main.nf"
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fasta/contigs.fasta', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fasta/scaffolds.fasta', checkIfExists: true)
                    ]
                ]
                """
            }
        }

        run("SEQHASHER_SEQHASHER") {
            script "../../seqhasher/main.nf"
            process {
                """
                input[0] = CAT_CAT.out.file_out
                input[1] = file("https://github.com/nf-microbe/modules/raw/refs/heads/master/bin/seq-hasher", checkIfExists: true)
                """
            }
        }
    }

    test("duplicated fasta.gz") {
        when {
            process {
                """
                input[0] = SEQHASHER_SEQHASHER.out.tsv
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = SEQHASHER_SEQHASHER.out.tsv
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
