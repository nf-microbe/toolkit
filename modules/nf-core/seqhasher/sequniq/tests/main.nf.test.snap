{
    "duplicated fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.dups.tsv:md5,063b88a9c4317628426c4cc0b12116df"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.seq-uniq.tsv:md5,063b88a9c4317628426c4cc0b12116df"
                    ]
                ],
                "2": [
                    "versions.yml:md5,0fdab31c01fc55b29ac740f577cb358d"
                ],
                "dups": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.dups.tsv:md5,063b88a9c4317628426c4cc0b12116df"
                    ]
                ],
                "unique": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.seq-uniq.tsv:md5,063b88a9c4317628426c4cc0b12116df"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0fdab31c01fc55b29ac740f577cb358d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-07T16:57:15.638844783"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.dups.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.seq-uniq.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,0fdab31c01fc55b29ac740f577cb358d"
                ],
                "dups": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.dups.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "unique": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.seq-uniq.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0fdab31c01fc55b29ac740f577cb358d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-07T16:57:42.882877318"
    }
}